doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+  - [Wildcard support](#wildcard-support)
+  - [Registration of Datapoints](#registration-of-datapoints)
+  - [Availability of Datapoints](#availability-of-datapoints)
+  - [Lifecycle of components](#lifecycle-of-components)
+  - [Path requests](#path-requests)
+  - [Errors](#errors)
+  - [Setting Values](#setting-values)
+  - [Atomic operatoins](#atomic-operatoins)
+  - [Update notifications](#update-notifications)
+  - [Access rights](#access-rights)
+  - [VSS signals - Users vs providers](#vss-signals---users-vs-providers)
+  - [Enable "easy to use" user facing API](#enable-easy-to-use-user-facing-api)
+  - [Performance / runtime footprint](#performance--runtime-footprint)
+  - [Throttling-mode when system is overloaded.](#throttling-mode-when-system-is-overloaded)
+  - [Considerations regarding shared and zero-copy memory approaches](#considerations-regarding-shared-and-zero-copy-memory-approaches)
+  - [Provider control and provider capabilities](#provider-control-and-provider-capabilities)
+  - [Control the rate of updates](#control-the-rate-of-updates)
+  - [Differentiate between different providers of the same VSS data](#differentiate-between-different-providers-of-the-same-vss-data)
+  - [Data Aliveness/Availability](#data-alivenessavailability)
+  - [Missing features from `sdv.databroker.v1` in `kuksa.val.v1`](#missing-features-from-sdvdatabrokerv1-in-kuksavalv1)
+- [Exploring a design of a bidirectional streaming API](#exploring-a-design-of-a-bidirectional-streaming-api)
+    - [Design choices](#design-choices)
+        - [Stream](#stream)
+        - [Bidirectional stream](#bidirectional-stream)
+        - [Actuators](#actuators)
+    - [Overview](#overview)
+  - [COVESA topics](#covesa-topics)
+- [Domain requirements](#domain-requirements)
+  - [Interface requirements](#interface-requirements)
+    - [Current Databroker API design](#current-databroker-api-design)
+    - [New Databroker API proposal](#new-databroker-api-proposal)
+- [Appendix](#appendix)
+  - [1. Use Cases](#1-use-cases)
+  - [2. Protobuf API Diagram](#2-protobuf-api-diagram)
+
+# Documentation
+  ### [Terminology](./terminology.md)
+  ### [System Architecture](./system-architecture.md)
+
+# Functional requirements
+
+### As Signal Consumer
+1. As a client I want to have a single service point accessible from my execution environment.
+   
+2. As a client I want to `Get` sensor values or actuator current values of the vehicle and its environment in a uniform manner.
+  - Details:
+    * All sensor and actuator values for which a client have permission to.
+    * Client must provide the paths of the requested signals, which optionally may contain wildcards.
+    * I want to receive exactly all the signals’ values I requested without errors.
+  
+3. As a client I want to `Subscribe` to sensor values or actuator current values when some signal value changed -> what about continuous subscription?
```

Comment:
I think we need to separate between at least three use-cases:

- Databroker has not received a new datapoint at all (from CAN, SOME/IP, ...)
- Databroker has received a new Datapoint (with new timestamp), but value is the same
- Databroker has received a new Datapoint (with new timestamp and value)

A typical use could be tire pressure monitoring, you may be interested in getting every observation even if value stays the same, as lack of observations might indicate a problem with the system.

Reply:
In addition to what @erikbosch writes, I would add to the sentence at least something like `to sensor values or actuator current values AND GET THEN NOTIFIED when some signal value changed`.

Reply:
For me there is also room for interpretation whether a client can subscribe to multiple signals and is then notified  with all signals when one signal changes or only with the changed signals. 

For instance, the client subscribes with a single call to A, B, and C. After B changes, does the client get only B or also A and C where the values remain the same. My intuition would say only B, but just reading the requirement both seems possible to me.